docs: add new read-only states pattern

laurenmrice opened the pull request:
Closes #3248 

Add a new pattern for read-only states with written guidance and images.

#### Changelog

**New**

- Content
- Images

alisonjoseph commented:
Looks great! Do we want to index this for Platform?

aagonzales commented on src/pages/patterns/read-only-states-pattern/index.mdx:
```diff
@@ -0,0 +1,360 @@
+1. **Background color change:** Having a transparent background for fields.
+2. **Border color change:** De-emphasizing selection and clickability to make
+   information more readable.
+3. **Icon color change:** Keepingg embedded icons in the component for context
```
```suggestion
3. **Icon color change:** Keeping embedded icons in the component for context
```

aagonzales commented on src/pages/patterns/read-only-states-pattern/index.mdx:
```diff
@@ -0,0 +1,360 @@
+2. **Border color change:** De-emphasizing selection and clickability to make
+   information more readable.
+3. **Icon color change:** Keepingg embedded icons in the component for context
+   but displaying that icons are not interactive with color and cursor changes.
```
I think we should add a 4th around the type
"Text color no change: text color remains the same as in the enabled state and should still pass 4.5:1 color contrast rules."

laurenmrice replied:
Good idea! Will add 👍

aagonzales commented on src/pages/patterns/read-only-states-pattern/index.mdx:
```diff
@@ -0,0 +1,360 @@
+### Structure
+
+Components should maintain the same structure and spacing used in the
+component’s enabled state. In most cases, elements in the enabled state should
+be present in the read-only state.
```
```suggestion
component’s enabled state. In most cases, elements in the enabled state are present in the read-only state.
```

aagonzales commented on src/pages/patterns/read-only-states-pattern/index.mdx:
```diff
@@ -0,0 +1,360 @@
+
+Selection controls offer users a selection from pre-determined options. Common
+selection controls include: checkboxes, radio buttons, toggles, dropdowns,
+selects, comboboxes and multiselects.
```
```suggestion
selects, combo boxes, and multiselects.
```

aagonzales commented on src/pages/patterns/read-only-states-pattern/index.mdx:
```diff
@@ -0,0 +1,360 @@
+### Interactive elements
+
+Interactive elements included in a component’s enabled state should be removed
+or modified for its read-only state. Changes to color can be used to modify
+interactive elements and change their focus from interactive to informative.
```
I would avoid the word "focus" outside the context of the focus state.

```suggestion
or modified for its read-only state. Changes to color are used to indicate a modification to
interactive elements and a change in their purpose from interactive to informative.
```

aagonzales commented on src/pages/patterns/read-only-states-pattern/index.mdx:
```diff
@@ -0,0 +1,360 @@
+#### Component icon signifiers
+
+Signifier icons included in a component’s enabled state, like chevron arrow
+icons, close icons, and calendar icons, should use the $icon-disabled color
```
```suggestion
icons, close icons, and calendar icons, should use the `$icon-disabled` color
```

aagonzales commented on src/pages/patterns/read-only-states-pattern/index.mdx:
```diff
@@ -0,0 +1,360 @@
+
+<Tab label="Read-only">
+
+![Read-only content for a dropdown with no selection.](images/read-only-content.png)
```
In this image, the field text color should still pass 4.5:1 contrast, change to text-secondary?
<img width="878" alt="image" src="https://user-images.githubusercontent.com/11670886/206320734-527c0f2e-cdf8-4581-bbe1-8edbab84dcbd.png">

laurenmrice replied:
Ok yes, I thought something was off here. I will try out `$text-secondary`.

aagonzales commented on src/pages/patterns/read-only-states-pattern/index.mdx:
```diff
@@ -0,0 +1,360 @@
+When considering keyboard accessibility,  it can be helpful to distinguish
+between navigation and operation. A component that can be reached by keyboard is
+navigable. Read-only components remain navigable so that users can review the
+information they contain. (This contrasts with disabled components, which cannot
+be reached by keyboard.) However, read-only components are not operable—users
```
That one sentence doesnt need the ( ) and I removed the em dash structure. 
```suggestion
information they contain. This contrasts with disabled components, which cannot be reached by a keyboard. However, read-only components are not operable, meaning users
```

aagonzales reviewed (changes requested):
This looks awesome, you did a great job bringing this across the finish line. I just had a few content suggestions to help with clarity in places.

laurenmrice commented on src/pages/patterns/read-only-states-pattern/index.mdx:
```diff
@@ -0,0 +1,360 @@
+<PageDescription>
+
+Read-only states are applied to components that users can review but not modify
+and remove a component’s interactive functions.
```
Agreed! Thank you.

laurenmrice commented:
@aagonzales Okay, all feedback has been added! Thank you.

— I update the read-only state colors with the latest spec for all images.
— I updated the do/don't image with a new UI.
— I also added the fluid variant in images when I am using the tab component for comparisons between the read-only/enabled states since they have some differences. Let me know what you think about that.

aagonzales commented on src/pages/patterns/read-only-states-pattern/index.mdx:
```diff
@@ -0,0 +1,365 @@
+<PageDescription>
+
+Read-only states are applied to components when the user is allowed to review
+but not modify the component. It removed all interactive functions from the
```
oops this is my own typo, should this be "removes" instead?
```suggestion
but not modify the component. It removes all interactive functions from the
```

laurenmrice replied:
Oh, yep! Good catch.

laurenmrice commented:
— Added a notification for fluid components.

aagonzales reviewed (approved):
Ship it!! Looks good!
